stl/inc/fstream

```diff
@@ -176,10 +176,6 @@ public:
     }
 
     __CLR_OR_THIS_CALL ~basic_filebuf() noexcept override {
-        if (_Myfile) {
-            _Reset_back(); // revert from _Mychar buffer
-        }

```

Comment:
I see how the addition on 381 fixes the bug, but I'm concerned that removing `_Reset_back()` here is creating a different bug for users of the `basic_filebuf(FILE*)` extension constructor. That constructor sets `_Myfile` but not `_Closef`, so if `_Set_back()` was ever evaluated I think we'll leave the `FILE` in a bad state after destroying `this` without evaluating `_Reset_back()`.

Am I missing something?

Reply:
I missed that there is a constructor that does not close the file :(